hw/top_earlgrey/data/top_earlgrey.c.tpl

```diff
@@ -7,11 +7,11 @@
 /**
  * PLIC Interrupt Id to Peripheral Map
  *
- * This array is a mapping from `top_earlgrey_plic_irq_id_t` to
- * `top_earlgrey_plic_peripheral_t`.
+ * This array is a mapping from `top_${top_name}_plic_irq_id_t` to
```

Comment:
You can use `top["name"]` everywhere in the templates that you use `top_name` - as this is what you do in `top_earlgrey.sv.tpl`. Then you don't need to pass another variable through `render_template`.